packages/invoices/src/InboundShipment/api/api.ts

```diff
@@ -195,26 +196,37 @@ export const getInboundQueries = (sdk: Sdk, storeId: string) => ({
       input: inboundParsers.toUpdate(patch),
       storeId,
     }),
-  deleteLines: async (invoiceId: string, ids: string[]) => {
-    const createDeleteLineInput = inboundParsers.toDeleteLine(invoiceId);
+  deleteLines: async (lines: { id: string; invoiceId: string }[]) => {
     return sdk.deleteInboundShipmentLines({
       storeId,
-      input: { deleteInboundShipmentLines: ids.map(createDeleteLineInput) },
+      input: {
+        deleteInboundShipmentLines: lines.map(inboundParsers.toDeleteLine),
+      },
     });
   },
   upsertLines: async (lines: DraftInboundLine[]) => {
     const insertInboundShipmentLines = lines
-      .filter(({ isCreated }) => isCreated)
+      .filter(({ isCreated, numberOfPacks }) => isCreated && numberOfPacks > 0)
       .map(inboundParsers.toInsertLine);
     const updateInboundShipmentLines = lines
-      .filter(({ isCreated, isUpdated }) => !isCreated && isUpdated)
+      .filter(
+        ({ isCreated, isUpdated, numberOfPacks }) =>
+          !isCreated && isUpdated && numberOfPacks > 0
+      )
       .map(inboundParsers.toUpdateLine);
+    const deleteInboundShipmentLines = lines
+      .filter(
+        ({ isCreated, isUpdated, numberOfPacks }) =>
+          !isCreated && isUpdated && numberOfPacks === 0
+      )
+      .map(inboundParsers.toDeleteLine);
```

Comment:
this small change here was really what the issue was about. Oops.

Delete lines which have been updated to zero quantity, insert only if more than `numberOfPacks` and only send an update if greater than zero packs (API errors otherwise)